Source/DafnyStandardLibraries/src/Std/Parsers/String/StringBuilders.dfy

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT 
+ *******************************************************************************/
+
+/** The parsers builder DSL but specialized for strings */
+
+module Std.Parsers.StringBuilders refines Builders {
+  import P = StringParsers
+  export StringBuilders extends Builders
+    provides ToInput, ToInputEnd, S, Int, Nat, WS, Except, Digit, DigitNumber, DebugSummaryInput, PrintDebugSummaryOutput, FailureToString, Apply, InputToString
+
+  function ToInput(other: seq<C>): (i: Input)
+    ensures P.A.View(i) == other
+  {
+    P.A.Input(other, 0, |other|)
+  }
+
+  function ToInputEnd(other: seq<C>, fromEnd: int := 0): (i: Input)
+    requires 0 <= fromEnd <= |other|
+    ensures P.A.View(i) == other[|other|-fromEnd..|other|]
+  {
+    P.A.Input(other, |other| - fromEnd, |other|)
+  }
+
+  /** `S(PREFIX)` returns success on its input if its starts with `PREFIX`,  
+      On success, `.value` contains the extracted string. You can ignore it by chaining it like `S(PREFIX).e_I(...)`  
+      On failure, returns a recoverable error that did not consume the input.*/
+  function S(s: string): B<string> {
```

Comment:
I feel like calling this `Fixed` or `Constant`...

Reply:
See [this comment](https://github.com/dafny-lang/dafny/pull/6143/#discussion_r2083115602) as I'm adding a description about why short names in parser combinators matter.
This include the parser combinator that test prefixes. We can definitely change the name in `ParserCore.dfy`, but for the parser builders the shortest is better, and `S` is an extremely used parser combinator, so it deserve a one-letter function.